pkg/cluster/service.go

```diff
@@ -0,0 +1,154 @@
+package cluster
+
+import (
+	"context"
+	"fmt"
+	"github.com/go-logr/logr"
+	"github.com/pkg/errors"
+	"k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/kubefed/pkg/apis/core/v1beta1"
+	"sigs.k8s.io/kubefed/pkg/controller/util"
+)
+
+const (
+	labelType      = "type"
+	labelNamespace = "namespace"
+
+	defaultHostOperatorNamespace   = "toolchain-host-operator"
+	defaultMemberOperatorNamespace = "toolchain-member-operator"
+)
+
+// KubeFedClusterService manages cached cluster kube clients and related KubeFedCluster CRDs
+// it's used for adding/updating/deleting
+type KubeFedClusterService struct {
+	Client client.Client
+	Log    logr.Logger
+}
+
+// AddKubeFedCluster takes the KubeFedCluster CR object,
+// creates FedCluster with a kube client and stores it in a cache
+func (s *KubeFedClusterService) AddKubeFedCluster(obj interface{}) {
+	cluster, ok := obj.(*v1beta1.KubeFedCluster)
```

Comment:
these first lines are repeated in every "exposed" methods of this service. Maybe you could factorize them?

Reply:
I was thinking about it when I was writing it, but I would safe max one line per method comparing to 7 new lines of the new function. That's why I decided to not do that...  But I see that it doesn't look good, so I changed it: 9147ad1